src/main/java/GooglePhotosAutoUploader.java

```diff
@@ -0,0 +1,31 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class GooglePhotosAutoUploader {
+    private Object LOCK = new Object();
+    private List<String> photosToUploadList = new ArrayList<>();
+    public void startAutoUpload() {
+        synchronized (LOCK) {
+            if (photosToUploadList.isEmpty()) {
+                try {
+                    LOCK.wait();
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e.getMessage());
+                }
+            }
+            uploadPhotos();
+        }
+    }
```

Comment:
Я бы предложил что-то в виде:

```suggestion
    public void startAutoUpload() {
        while (true) {
            synchronized (LOCK) {
                while (photosToUpload.isEmpty()) {
                    try {
                        LOCK.wait();
                    } catch (InterruptedException e) {
                        log.error("Ошибка при ожидании новых фотографий");
                        Thread.currentThread().interrupt();
                        return;
                    }
                }
                uploadPhotos();
            }
        }
    }
```